core/domain/stats_services.py

```diff
@@ -193,117 +139,99 @@ def get_visualizations_info(exploration_id, exploration_version, state_name):
         calculation_ids_to_outputs[calculation_id] = (
             calc_output_domain_object.calculation_output)
 
-    results_list = [{
+    return [{
         'id': visualization.id,
         'data': calculation_ids_to_outputs[visualization.calculation_id],
         'options': visualization.options,
     } for visualization in visualizations
-                    if visualization.calculation_id
-                    in calculation_ids_to_outputs]
-
-    return results_list
+            if visualization.calculation_id in calculation_ids_to_outputs]
 
 
-def get_top_state_rule_answers(exploration_id, state_name, rule_str_list):
-    """Returns a list of top answers (by submission frequency) submitted to the
-    given state in the given exploration which were mapped to any of the rules
-    listed in 'rule_str_list'. All answers submitted to the specified state and
-    match the rule spec strings in rule_str_list are returned.
+def get_top_state_rule_answers(
+        exploration_id, state_name, classify_category_list):
```

Comment:
It's actually quite simple; updated the docstring to illustrate why. exp_domain.EXPLICIT_CLASSIFICATION is a classification category. Do you still think the names need to be changed? I think it's clear once that's established (here and elsewhere).